keras_cv/applications/object_detection/retina_net/__internal__/utils/label_encoder.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+from keras_cv import bounding_box
+from keras_cv.applications.object_detection.retina_net.__internal__ import utils
+
+
+class LabelEncoder(layers.Layer):
+    """Transforms the raw labels into targets for training.
+
+    This class has operations to generate targets for a batch of samples which
+    is made up of the input images, bounding boxes for the objects present and
+    their class ids.
+
+    Args:
+        bounding_box_format:  The format of bounding boxes of input dataset. Refer
+            https://github.com/keras-team/keras-cv/blob/master/keras_cv/bounding_box/converters.py
+            for more details on supported bounding box formats.
+        anchor_box: Anchor box generator to encode the bounding boxes.
+        box_variance: The scaling factors used to scale the bounding box targets.
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        anchor_box_generator=None,
+        box_variance=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self._anchor_box = anchor_box_generator or utils.AnchorBox(
+            bounding_box_format=bounding_box_format
+        )
+
+        box_variance = box_variance or [0.1, 0.1, 0.2, 0.2]
+        self._box_variance = tf.convert_to_tensor(box_variance, dtype=tf.float32)
+
+    def _match_anchor_boxes(
+        self, anchor_boxes, gt_boxes, match_iou=0.5, ignore_iou=0.4
```

Comment:
Unless there is any reason for the mathc_iou and ignore_iou to have default value here, I think we should expose them to the layer as the init args.

Reply:
Lets do this in the PR where we make label_encoder a real API.  Zhenyu has a good implementation of this that MG uses, which we plan to port in a follow up.